Swiftfin/Views/SettingsView/UserDashboardView/UsersView/UsersView.swift

```diff
@@ -0,0 +1,90 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2024 Jellyfin & Jellyfin Contributors
+//
+
+import CollectionVGrid
+import Defaults
+import JellyfinAPI
+import SwiftUI
+
+struct UsersView: View {
+
+    @EnvironmentObject
+    private var router: SettingsCoordinator.Router
+
+    @StateObject
+    private var viewModel = UsersViewModel()
+
+    // MARK: - Body
+
+    var body: some View {
+        ZStack {
+            switch viewModel.state {
+            case .content:
+                if viewModel.users.isEmpty {
+                    Text(L10n.none)
+                } else {
+                    contentView
+                }
+            case let .error(error):
+                errorView(with: error)
+            case .initial:
+                DelayedProgressView()
+            }
+        }
+        .navigationTitle(L10n.users)
+        .onFirstAppear {
+            viewModel.send(.getUsers)
+        }
+        .refreshable {
+            viewModel.send(.getUsers)
+        }
+        .topBarTrailing {
```

Comment:
We may want to retrieve users based on the hidden/disabled filters and have a filter for that, if we want to do that now.

Reply:
Sorry, I don't think I fully understand what you mean. Do you mean add an action for getDisabledUsers / getHiddenUsers? Or add an extra parameter to the getUsers action like getUsers(active: true, inactive: false)?

Reply:
The latter. We can just have 2 `State` variables for the filter states and are set through a `Menu`.

Reply:
This is giving me some trouble since we have a state enum in the same struct. Is there a way to force this so it knows I mean the properly NOT the enum?
<img width="309" alt="Screenshot 2024-10-26 at 22 44 24" src="https://github.com/user-attachments/assets/bd38eceb-d214-4b88-900f-6ffbefd445ca">

Reply:
I am unsure of what exactly that is coming from. I mean having two `@State` variables within the view like below:

```swift
struct ... : View {

	@State
	private var isDisabledFilterActive = false
	@State
	private var isHiddenFilterActive = false
}
```

When these are updated, we update the device list.

Reply:
Ohhhh..... I misunderstood. I was trying to put states on the viewModel... Sorry about that! I know what direction to go now

Reply:
All right! I have this configured. This works how I would expect it and I have this defaulted to include all users regardless of hidden/disabled status. I think the only outlier now is making sure the filter is placed in a location that makes sense. I currently have it where it just sits on top of the user list.